### PR TITLE
Qual 215 upgrade nest package

### PR DESCRIPTION
**Pull request by CodeWolf86**

### CODE REVIEW

File: SFA.DAS.VacancyServices.Search/SFA.DAS.VacancyServices.Search/ElasticSearchFactory.cs [hunk 1]
```diff
@@ -1,16 +1,32 @@
...
 {
     using System;
+    using Elasticsearch.Net;
     using Nest;
     using Newtonsoft.Json.Converters;
 
     internal class ElasticSearchFactory : IElasticSearchFactory
     {
-        public IElasticClient GetElasticClient(string hostName)
+        public IElasticClient GetElasticClient(ApprenticeshipSearchClientConfiguration config)
```
Comment (dev4valtech) [inline SFA.DAS.VacancyServices.Search/SFA.DAS.VacancyServices.Search/ElasticSearchFactory.cs]: I think we need to refactor this per my PR in FAA. Infact if possible get rid of the factory from FAA and reuse this instead. Need to check impact on the API

File: SFA.DAS.VacancyServices.Search/SFA.DAS.VacancyServices.Search/ElasticSearchFactory.cs [hunk 2]
```diff
@@ -1,16 +1,32 @@
...
         {
-            var elasticConnectionSettings = new ConnectionSettings(new Uri(hostName));
-            elasticConnectionSettings.AddContractJsonConverters(t => typeof(Enum).IsAssignableFrom(t) ? new StringEnumConverter() : null);
+            return GetElasticClient(config.HostName, config.Username, config.Password);
+        }
+
+        public IElasticClient GetElasticClient(TraineeshipSearchClientConfiguration config)
```
Comment (dev4valtech) [inline SFA.DAS.VacancyServices.Search/SFA.DAS.VacancyServices.Search/ElasticSearchFactory.cs]: Also there is only going to be one single config for all the indexes.

File: SFA.DAS.VacancyServices.Search/SFA.DAS.VacancyServices.Search/Extensions/SearchDescriptorExtensions.cs [hunk 3]
```diff
@@ -7,34 +7,34 @@ namespace SFA.DAS.VacancyServices.Search
...
-                {
-                    g.PinTo(searchParameters.Latitude.Value, searchParameters.Longitude.Value)
-                        .Unit(GeoUnit.Miles).OnField(f => f.Location);
-                    return g;
-                });
+                sortDescriptor.GeoDistance(g => g
+                .Field(f => f.Location)
+                .DistanceType(GeoDistanceType.Arc)
+                .Unit(DistanceUnit.Miles)
+                .Mode(SortMode.Min)
```
Comment (dev4valtech) [inline SFA.DAS.VacancyServices.Search/SFA.DAS.VacancyServices.Search/Extensions/SearchDescriptorExtensions.cs]: I thought you took mode out as it is not applicable?

File: SFA.DAS.VacancyServices.Search/SFA.DAS.VacancyServices.Search/TraineeshipSearchClient.cs [hunk 4]
```diff
@@ -68,15 +66,14 @@ public TraineeshipSearchResponse Search(TraineeshipSearchRequestParameters searc
...
         {
-            var client = _elasticSearchFactory.GetElasticClient(_config.HostName);
+            var client = _elasticSearchFactory.GetElasticClient(_config);
 
             var results = client.Search<TraineeshipSearchResult>(s =>
             {
                 s.Index(_config.Index);
-                s.Type(ElasticTypes.Traineeship);
                 s.Skip((parameters.PageNumber - 1) * parameters.PageSize);
                 s.Take(parameters.PageSize);
```
Comment (dev4valtech) [inline SFA.DAS.VacancyServices.Search/SFA.DAS.VacancyServices.Search/TraineeshipSearchClient.cs]: I had included s.Source(); here to retain fields, but from what I read this is enabled by default in 7.x so hopefully we are alright. I think result comparison will help in this case.

File: SFA.DAS.VacancyServices.Search/SFA.DAS.VacancyServices.Search/Extensions/SearchDescriptorExtensions.cs (hunk 3 above)
Comment (CodeWolf86) [reply SFA.DAS.VacancyServices.Search/SFA.DAS.VacancyServices.Search/Extensions/SearchDescriptorExtensions.cs]: Yep hadn't got around to sorting that yet. Its the default mode so doesn't need to be specified. Have pushed now.

**Pull request merged**